### PR TITLE
Follow up PR for Audio End to End testing

### PR DESCRIPTION
**Pull request by wirthual**
#378 This PR adds support for handing text and urls for the audio embeddings endpoint.

Extended the test to test text only, mixed text and urls and also check a meta test for the embeddings.

Implemented for the audio endpoint, if this scheme is acceptable happy to port it to the vision endpoint.

Cheers

### CODE REVIEW

**Comment by wirthual** [comment]
> @wirthual Wait, I am confused - the “/embeddings” endpoint already works for audio!

Isn`t the `/embeddings` endpoint for text only and the `/embeddings_audio` for the clap model?

File: libs/infinity_emb/infinity_emb/infinity_server.py [hunk 1]
```diff
@@ -411,29 +412,54 @@ async def _embeddings_audio(data: AudioEmbeddingInput):
...
+            logger.debug(
+                f"[📝] Received request with {len(audio_urls)} Urls and {len(texts)} sentences"
+            )
             start = time.perf_counter()
 
-            embedding, usage = await engine.audio_embed(audios=audio_inputs)  # type: ignore
+            if audio_urls:
+                audio_embeddings, usage = await engine.audio_embed(audios=audio_urls)  # type: ignore
+            if texts:
+                text_embeddings, usage = await engine.embed(sentences=texts)
```
[the remark this answers is not in the document]
Comment (wirthual) [reply libs/infinity_emb/infinity_emb/infinity_server.py]: Not valid anymore

File: libs/infinity_emb/tests/end_to_end/test_torch_audio.py [hunk 2]
```diff
@@ -79,6 +85,54 @@ async def test_audio_single_text_only(client):
...
+    embeddings_audio_beep = rdata_results[0]["embedding"]
+    embeddings_text_beep = rdata_results[1]["embedding"]
+    embeddings_text_horse = rdata_results[2]["embedding"]
+    embeddings_text_fish = rdata_results[3]["embedding"]
+    assert cosine_similarity(
+        embeddings_audio_beep, embeddings_text_beep
+    ) > cosine_similarity(embeddings_audio_beep, embeddings_text_fish)
+    assert cosine_similarity(
+        embeddings_audio_beep, embeddings_text_beep
+    ) > cosine_similarity(embeddings_audio_beep, embeddings_text_horse)
```
[the remark this answers is not in the document]
Comment (wirthual) [reply libs/infinity_emb/tests/end_to_end/test_torch_audio.py]: Just a greater than is fine here I guess

**Review by michaelfeil — APPROVED** [review]
LGTM, thanks for the contribution! Feel free to merge once all test are passing.

**Pull request merged**